sparse/sfold.go

```diff
@@ -63,25 +63,33 @@ func coalesce(parentFileIo FileIoProcessor, childFileIo FileIoProcessor) error {
 		}
 
 		// now we have a data start offset and length(hole - data)
-		// let's read from child and write to parent file block by block
+		// let's read from child and write to parent file. We read/write up to
+		// 32 blocks in a batch
 		_, err = parentFileIo.Seek(data, os.SEEK_SET)
 		if err != nil {
 			log.Error("Failed to os.Seek os.SEEK_SET")
 			return err
 		}
 
-		offset := data
-		buffer := AllocateAligned(blockSize)
-		for offset != hole {
-			// read a block from child, maybe use bufio or Reader stream
-			n, err := childFileIo.fileReadAt(buffer, offset)
-			if n != len(buffer) || err != nil {
+		batch := 32 * blockSize
+		if hole-data < int64(batch) {
+			batch = int((hole - data) * 32)
```

Comment:
Good catch. No one will understand it, including me :-( It is a typo(*32 part).  I should have double checked this before sending out, my bad.